tests/core/emscripten_lazy_load_code.cpp

```diff
@@ -0,0 +1,43 @@
+#include <emscripten.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+extern "C" {
+
+EM_JS(void, log_started, (), {
+  out("foo_start");
+})
+
+EM_JS(void, log_ended, (), {
+  out("foo_end");
+})
+
+void foo_start(int n) {
+  log_started();
+  // prevent inlining
+  if (n > 10000) foo_start(n - 1);
+  if (n > 15000) foo_start(n - 3);
+}
+
+void foo_end(int n) {
+  log_ended();
+  // prevent inlining
```

Comment:
How about `__attribute__((noinline))` to be explicit?

Reply:
That's not enough for binaryen, though :(